galil_ini_parser/galil_ini_parser.py

```diff
@@ -0,0 +1,219 @@
+        settings.append("[{}]".format(self.crate_index))
+        for crate_setting, setting_value in self.settings.items():
+            settings.append("{setting} = {value}".format(setting=crate_setting, value=setting_value))
+        for axis_letter in self.axes.keys():
+            axis = self.axes[axis_letter]
+            for setting, value in axis.settings.items():
+                settings.append("Axis {axis_letter} {setting} = {value}".format(axis_letter=axis_letter,
+                                                                                setting=setting,
+                                                                                value=value))
+        return settings
+
+
+class Axis:
+    def __init__(self, axis_index: str):
+        self.settings = {}
+        self.axis_index = axis_index
+        self.axis_line_prefix = "Axis {}".format(axis_index)
+
+    def scrub_axis_prefix(self, ini_line: str) -> str:
+        """
+        Removes the axis identifier from a line in the ini file. Returns a string which defines its setting
+
+        Args:
+            ini_line: Line from the ini file
+
+        Returns:
+            setting: String containing "setting_name = value" pair 
+        """
+        # Scrub the Axis prefix from the line
+        setting = ini_line[len(self.axis_line_prefix):].strip()
+        return setting
+
+    def add_setting_from_ini_line(self, ini_line: str):
+        """
+        Imports a new setting from a line in the galil ini file
+        """
+        setting = self.scrub_axis_prefix(ini_line)
+        split_line = setting.split("=")
+        key = split_line[0].strip()
+
+        # This captures settings which contain an equals sign in them
+        value = "=".join(split_line[1:]).strip()
+        self.set_value(key, value, make_new=True)
+
+    def get_value(self, setting: str, caster: Any) -> Optional[Any]:
+        """
+        Attempts to get the setting for this axis and cast it to type 'type'.
+
+        Args:
+            setting: The name of the setting to get
+            type: Function which casts string to expected type of setting
+
+        Returns:
+            value: The setting, cast to 'type'. Is None if the setting does not exist for this axis
+        """
+
+        try:
+            value = caster(self.settings[setting])
+        except KeyError:
+            value = None
+
+        return value
+
+    def set_value(self, setting: str, value: str, make_new=False) -> None:
+        """
+        Sets the setting with the specified value
+
+        Args:
+            setting: The setting to set
+            value: A string representation of the value to set
+            make_new: If True, create a new setting even if it does not currently exist
+        """
+        if setting in self.settings.keys():
+            self.settings[setting] = value
+        elif make_new:
+            self.settings[setting] = value
+
+    def get_motor_resolution(self) -> float:
```

Comment:
It's not just getting the motor resolution, rename to something like get smallest movement?